.github/workflows/update_python_dependencies.yaml

```diff
@@ -0,0 +1,67 @@
+---
+name: Update Python dependencies
+
+# Run workflow on pushes to matching branches
+on:  # yamllint disable-line rule:truthy
+  push:
+    branches: [develop]
+  schedule:
+    - cron: "0 3 * * 1"  # run at 3:00 every Monday
+  workflow_dispatch:  # allow this workflow to be manually triggered
+
+
+jobs:
+  update_python_dependencies:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: 3.12
+
+      - name: Install dependencies
+        run: pip install hatch pip-tools
+
+      - name: Update 'default' dependencies
+        run: pip-compile -U pyproject.toml -o requirements.txt
+
+      - name: Update 'docs' dependencies
+        run: hatch env show docs --json | jq -r '.docs.dependencies | .[]' | pip-compile -U - -o docs/requirements.txt
```

Comment:
I think this was one of the reasons I went with a `requirements.in`, because there isn't a convenient command like `pip-compile pyproject.toml --env docs`.

If we could make that a script rather than being hardcoded in the workflow that could be good.

Reply:
I think the main reason for going with `requirements.in` here was that we didn't have a top-level `pyproject.toml` as this workflow existed in the Powershell version.

There is a [hatch-pip-compile](https://github.com/juftin/hatch-pip-compile) plugin, but we need the generation of the `requirements` file to happen outside `hatch`, as one of the hatch steps is to install from this file!

Reply:
> I think the main reason for going with requirements.in here was that we didn't have a top-level pyproject.toml as this workflow existed in the Powershell version.

That makes sense :smile:.

That plugin sounds a bit awkward. If that chain of commands were just in a file like `update_docs_requirements.sh` I just think it would be more visible, easier to reproduce out of CI.

Reply:
I don't think the overhead of moving this into its own script is worthwhile. It would make sense if we were doing so for more than one environment though.